mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -1916,6 +1949,10 @@ fun View.adjustForWidgetHeight(widget: Widget, fallbackRowCount: Int) {
     }
 }
 
+fun View.setReadOnly(widget: Widget) {
```

Comment:
How about, instead of this extension, giving `Widget` a property?
`val isReadOnly get() = item?.readOnly == true`

That property can be used everywhere, unlike the extension method.